controllers/userController.ts

```diff
@@ -51,3 +51,38 @@ export const update =
             return next(error);
         }
     };
+
+// Delete controller
+// ----------------------------------------------------------------------------
+
+export const deleteUser =
+    (deleteUser: UserMethods.DeleteUser): RequestHandler =>
+    async (
+        req: Request,
+        res: Response,
+        next: NextFunction
+    ): Promise<void | Response> => {
+        try {
+            // Check if the request has a session
+            if (req?.session?.user?.id) {
+                const result = await deleteUser(req?.session?.user?.id);
+
+                // If the user exists, return a status 204:
+                if (result) {
+                    return res.status(204).json();
+                }
+
+                // Else, if the user doesn't exist, return a status 400:
+                if (!result) {
+                    return res.status(400).json({
+                        status: "error",
+                        data: `User could not be found, ${req?.session?.user?.id} is not valid`,
+                    });
+                }
+            }
```

Comment:
This actually just checks whether or not the user was deleted. It will work if the operation was successful, but will throw an error if it fails. The `if(!result)` conditional won't catch an error. We need to do an actual user existence check. To clarify my previous comment on this, the first method could be to do the user existence check in the controller:
```
// I would suggest storing the userId in a variable as it is used more than once
const userId = req?.session?.user?.id;
// User existence check
const userExists = await getUserById(userId);
if(!userExists) return res.status(400).json(..........);
// Run deletion query safely as the user definitely exists, if there is an error now then we do want to know about it
```
The other method is to run the same query in the model and return null. If you do this then what you currently have in the controller will work as expected as no error will be thrown if the user exists - it will return null and fail the conditional check as expected.

***Side Note***: Essentially this is an issue with the design of Prisma. Most databases, ORMs and ODMs will return null when the record doesn't exist. For some reason the Prisma team chose to throw an exception - so now we need to handle it.

***Opinion Piece***: Having used both Mongoose and Prisma I feel that writing queries directly has many advantages and little overhead. Mongoose makes more sense though as it makes MongoDB easier to use. I think that SQL based relational DBs may be more straight forward to just write the queries to be honest.